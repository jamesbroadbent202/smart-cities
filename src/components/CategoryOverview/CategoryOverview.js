import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import IndicatorCard from '../Card/IndicatorCard/IndicatorCard';
import Icon from '../Icon/Icon';
import Pill from '../Pill/Pill';
import aggregateIndicatorForCities from '../../helpers/aggregateIndicatorForCities';
import getColorVariant from '../../helpers/getColorVariant';
import {
  CATEGORY_IDS,
  INDICATORS,
  STRINGS,
} from '../../constants';
import COLORS from '../../style/_colors.scss';
import style from './CategoryOverview.scss';

const classnames = require('classnames/bind').bind(style);

const CategoryOverview = (props) => {
  const isContextCategory = props.category.id === CATEGORY_IDS.CONTEXT;
  const indicatorValue = props.city
    ? props.city.indices[props.category.heroIndicatorId]
    : aggregateIndicatorForCities(
      props.category.heroIndicatorId,
      props.cities,
    );

  const highlightColor = getColorVariant(props.category.colorName, '500');
  const categoryLinkStyle = isContextCategory
    ? {
      background: COLORS.WHITE,
      border: `2px solid ${highlightColor}`,
      color: COLORS.GREY_700,
    }
    : {
      background: highlightColor,
      border: `2px solid ${highlightColor}`,
      color: COLORS.WHITE,
    };

  const className = classnames(
    style.categoryWrapper,
    { category__context: isContextCategory },
  );

  return (
    <div
      className={className}
      style={{
        background: getColorVariant(props.category.colorName, '040'),
      }}
    >
      <div className={style.category}>
        {!isContextCategory && (
          <div className={style.categoryIcon}>
            <Icon
              icon={props.category.iconId}
              size={100}
              color={getColorVariant(props.category.colorName, '500')}
            />
          </div>
        )}

        <div className={style.categoryTextWrapper}>
          <h2 className={style.categoryTitle}>{props.category.name}</h2>

          <p className={style.categoryDescription}>{props.category.description}</p>

          <p className={style.subCategoryWrapperText}>{STRINGS.SUB_CATS_INCLUDE}</p>

          <div>
            {props.category.subCategories.map(subCategory => (
              <NavLink
                key={subCategory.name}
                to="TODO"
              >
                <Pill className={style.subCategoryLink} shadow>
                  {subCategory.name}
                </Pill>
              </NavLink>
            ))}
          </div>
        </div>

        <div className={style.indicatorCardAndLink}>
          <IndicatorCard
            className={style.indicatorCard}
            color={getColorVariant(props.category.colorName, '500')}
            colorName={props.category.colorName}
            indicator={props.category.heroIndicatorId}
            value={indicatorValue}
            showPerformanceIndicator={false}
          />

          <NavLink to="TODO">
            <Pill
              className={style.categoryLinkWrapper}
              style={categoryLinkStyle}
            >
              <span className={style.categoryLinkText}>
                View {props.category.name} section
              </span>

              <Icon
                icon={isContextCategory ? 'rightArrowInCircle' : 'rightArrowInCircleInverted'}
                size={20}
                color={getColorVariant(props.category.colorName, '500')}
              />
            </Pill>
          </NavLink>
        </div>
      </div>
    </div>
  );
};

const cityPropShape = {
  id: PropTypes.string.isRequired,
  indices: PropTypes.object.isRequired,
};

CategoryOverview.propTypes = {
  category: PropTypes.shape({
    colorName: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    heroIndicatorId: PropTypes.oneOf(Object.keys(INDICATORS)).isRequired,
    id: PropTypes.string.isRequired,
    iconId: PropTypes.string,
    name: PropTypes.string.isRequired,
    subCategories: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string.isRequired,
    })),
  }).isRequired,
  city: PropTypes.shape(cityPropShape),
  cities: PropTypes.arrayOf(PropTypes.shape(cityPropShape)),
  className: PropTypes.string,
};

export default CategoryOverview;
