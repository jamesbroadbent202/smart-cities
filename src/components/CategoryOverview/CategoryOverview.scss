@import '~style/variables';

.categoryWrapper {
  border-top: 8px solid $color-white;
}

.category {
  @include container;

  display: flex;
  flex-direction: column;
  padding-top: 32px;
  padding-bottom: 32px;

  @include for-tablet-portrait-up {
    padding-bottom: 38px;
    flex-direction: row;
  }
}

.categoryIcon {
  display: none;

  @include for-tablet-portrait-up {
    flex: none;
    width: 80px;
    margin-right: 32px;
    align-self: center;
    display: block;
  }

  @include for-tablet-landscape-up {
    width: 115px;
    margin-right: 40px;
  }
}

.textAndIndicatorWrapper {
  display: flex;
  flex-direction: column;

  @include for-tablet-landscape-up {
    flex-direction: row;
  }
}

.categoryTextWrapper {
  margin-bottom: 16px;

  @include for-tablet-portrait-up {
    margin-bottom: 32px;
  }

  @include for-tablet-landscape-up {
    margin-right: 32px;
    margin-bottom: 0;
  }
}

.categoryTitle {
  font-size: 34px;
  line-height: 1;
  margin-bottom: 16px;

  .category__context & {
    font-size: 20px;
    font-weight: 700;
  }

  @include for-tablet-landscape-up {
    font-size: 44px;

    .category__context & {
      font-size: 22px;
    }
  }
}

.categoryDescription {
  margin-bottom: 16px;
  font-size: 18px;
  line-height: 1.5;

  @include for-tablet-landscape-up {
    font-size: 22px;
  }
}

.subCategoryWrapperText {
  font-size: 12px;
  font-weight: 700;
  margin-bottom: 16px;
}

.subCategoryLink {
  margin-right: 8px;
  margin-bottom: 8px;
  padding: 0 16px;
  font-size: 14px;
  background: $color-white;

  &:hover {
    @include shadow(2);
  }

  @include for-tablet-portrait-up {
    padding: 0 24px;
    margin-right: 16px;
  }

  @include for-big-desktop-up {
    padding: 0 40px;
  }
}

.indicatorCardAndLink,
.indicatorCards {
  flex: none;
  display: flex;
  flex-direction: column;
  justify-content: space-between;

  @include for-tablet-portrait-up {
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
  }

  @include for-tablet-landscape-up {
    flex-direction: column;
    width: 256px;
    margin-left: auto;
  }
}

.indicatorCards {
  justify-content: center;

  @include for-tablet-landscape-up {
    flex: auto;
  }
}

.indicatorCard {
  margin-bottom: 16px;

  @include for-tablet-portrait-up {
    flex: 1;
    max-width: 256px;
    margin-bottom: 0;
    margin-top: 0;
    margin-right: 16px;
  }

  @include for-tablet-landscape-up {
    flex: none;
    width: 100%;
    margin-right: 0;
    margin-top: 8px;
  }
}

.categoryLinkWrapper {
  flex: 1;
  width: 100%;

  @include for-tablet-portrait-up {
    max-width: 256px;
  }

  @include for-tablet-landscape-up {
    flex: none;
    margin-top: 16px;
    justify-self: flex-end;
  }
}

.categoryLinkPill {
  display: flex;
  align-items: center;
  width: 100%;
  padding: 4px 3px 4px 0;
  border-radius: 10px;

  @include for-tablet-portrait-up {
    width: 100%;
    max-width: 256px;
  }

  @include for-tablet-landscape-up {
    width: 100%;
  }

  &:hover {
    @include shadow(2);
  }
}

.categoryLinkText {
  flex: auto;
  text-align: center;
  font-size: 14px;
}

.categoryLinkIcon {
  flex: none;
}
