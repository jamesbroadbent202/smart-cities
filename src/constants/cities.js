/* eslint-disable */

// This data is used when converting the source data to a format for use in the front end
// The key for each city must match exactly what appears in the source data
// An error will be thrown during conversion if they don't match
export default {
  'Albury - Wodonga': {
    id: 'albury-wodonga',
    shortName: 'Albury/Wodonga',
    description: 'TODO enter in a description for Albury - Wodonga',
  },
  'Australian Capital Territory': {
    id: 'act',
    shortName: 'ACT',
    description: 'TODO enter in a description for Australian Capital Territory',
  },
  'Ballarat': {
    id: 'ballarat',
    shortName: 'Ballarat',
    description: 'TODO enter in a description for Ballarat',
  },
  'Bendigo': {
    id: 'bendigo',
    shortName: 'Bendigo',
    description: 'TODO enter in a description for Bendigo',
  },
  'Cairns': {
    id: 'cairns',
    shortName: 'Cairns',
    description: 'TODO enter in a description for Cairns',
  },
  'Geelong': {
    id: 'geelong',
    shortName: 'Geelong',
    description: 'TODO enter in a description for Geelong',
  },
  'Gold Coast - Tweed Heads': {
    id: 'gold-coast-tweed-heads',
    shortName: 'Gold Coast/Tweed Heads',
    description: 'TODO enter in a description for Gold Coast - Tweed Heads',
  },
  'Greater Adelaide': {
    id: 'greater-adelaide',
    shortName: 'Adelaide',
    description: 'TODO enter in a description for Greater Adelaide',
  },
  'Greater Brisbane': {
    id: 'greater-brisbane',
    shortName: 'Brisbane',
    description: 'TODO enter in a description for Greater Brisbane',
  },
  'Greater Darwin': {
    id: 'greater-darwin',
    shortName: 'Darwin',
    description: 'TODO enter in a description for Greater Darwin',
  },
  'Greater Hobart': {
    id: 'greater-hobart',
    shortName: 'Hobart',
    description: 'TODO enter in a description for Greater Hobart',
  },
  'Greater Melbourne': {
    id: 'greater-melbourne',
    shortName: 'Melbourne',
    description: 'TODO enter in a description for Greater Melbourne',
  },
  'Greater Perth': {
    id: 'greater-perth',
    shortName: 'Perth',
    description: 'Perth is Western Australiaâ€™s state capital city. Perth has an estimated population of over 2.04 million people, which account for 8.58% of the national population. Perth is Australia 4th largest city in population size. Perth is part of the South West Land Division of Western Australia, with the majority of the metropolitan area of Perth located on the Swan Coastal Plain.',
  },
  'Greater Sydney': {
    id: 'greater-sydney',
    shortName: 'Sydney',
    description: 'TODO enter in a description for Greater Sydney',
  },
  'Launceston': {
    id: 'launceston',
    shortName: 'Launceston',
    description: 'TODO enter in a description for Launceston',
  },
  'Mackay': {
    id: 'mackay',
    shortName: 'Mackay',
    description: 'TODO enter in a description for Mackay',
  },
  'Newcastle - Maitland': {
    id: 'newcastle-maitland',
    shortName: 'Newcastle/Maitland',
    description: 'TODO enter in a description for Newcastle - Maitland',
  },
  'Sunshine Coast': {
    id: 'sunshine-coast',
    shortName: 'Sunshine Coast',
    description: 'TODO enter in a description for Sunshine Coast',
  },
  'Toowoomba': {
    id: 'toowoomba',
    shortName: 'Toowoomba',
    description: 'TODO enter in a description for Toowoomba',
  },
  'Townsville': {
    id: 'townsville',
    shortName: 'Townsville',
    description: 'TODO enter in a description for Townsville',
  },
  'Western Sydney': {
    id: 'western-sydney',
    shortName: 'Western Sydney',
    description: 'TODO enter in a description for Western Sydney',
  },
  'Wollongong': {
    id: 'wollongong',
    shortName: 'Wollongong',
    description: 'TODO enter in a description for Wollongong',
  },
};
