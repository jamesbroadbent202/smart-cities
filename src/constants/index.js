export * from './misc';
export { default as INDICATORS } from './indicators';
export { default as CATEGORIES } from './categories';
