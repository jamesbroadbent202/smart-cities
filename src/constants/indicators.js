import {
  DATA_TYPES,
  AGGREGATION_METHODS,
} from './misc';

// Different length text is used like so:
// name: shows in an indicator card and as the
// title of a chart (or in the legend for stacked charts)
// shortDescription: shows on the axis of a chart
// longDescription: shows in the tooltip for a chart
export default {
  participationRate: {
    name: 'Participation rate',
    shortDescription: 'TODO',
    longDescription: 'TODO',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.MEDIAN,
    format: '0[.]0%',
    source: 'Participation rate, % of population aged over 15 years',
  },
  gdp: {
    name: 'Economic output per capita',
    shortDescription: 'TODO',
    longDescription: 'TODO',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.MEDIAN,
    format: '$0,0.0a',
    source: 'Gross domestic product, AUD per capita per year',
  },
  educationalAttainmentYr12: {
    name: 'Educational attainment - year 12',
    shortDescription: 'TODO',
    longDescription: 'TODO',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.MEDIAN,
    format: '0[.]0%',
    source: 'Education attainment - school completion rate, Year 12 or equivalent, % of population',
  },
  population: {
    name: 'The population',
    shortDescription: 'TODO',
    longDescription: 'TODO',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.SUM,
    format: '0[.]0a',
    source: 'Population, All persons, People',
  },
  growthRate: {
    name: 'The growthRate',
    shortDescription: 'TODO',
    longDescription: 'TODO',
    cardSuffix: '/year',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.MEDIAN,
    format: '0[.]0%',
    source: 'Population growth rate, All persons, % per year',
  },
  dispersion: {
    name: 'Local government dispersion',
    shortDescription: 'Local government dispersion, Share of LGAs in bottom 10% by population',
    longDescription: 'Local government dispersion, Share of LGAs in bottom 10% by population',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.MEDIAN,
    format: '0[.]0%',
    source: 'Local government dispersion, Share of LGAs in bottom 10% by population',
  },
  roadCongestionCost: {
    name: 'Annual cost of road congestion',
    shortDescription: 'Annual cost of road congestion, AUD Billions per year, indexed to 2010',
    longDescription: 'Annual cost of road congestion, AUD Billions per year, indexed to 2010',
    cardSuffix: 'AUD',
    chartPrefix: '$',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.SUM,
    format: '$0,0[.]0a',
    source: 'Annual cost of road congestion , AUD Billions per year, indexed to 2010',
  },
  dwellingHouse: {
    name: 'House',
    shortDescription: 'TODO',
    longDescription: 'TODO',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.MEDIAN,
    format: '0[.]0%',
    source: 'Dwelling type ratio, Separate house, Percentage of Dwellings',
  },
  dwellingSemi: {
    name: 'Semi-House',
    shortDescription: 'TODO',
    longDescription: 'TODO',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.MEDIAN,
    format: '0[.]0%',
    source: 'Dwelling type ratio, Semi-detached, Percentage of Dwellings',
  },
  dwellingApartment: {
    name: 'Apartment',
    shortDescription: 'TODO',
    longDescription: 'TODO',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.MEDIAN,
    format: '0[.]0%',
    source: 'Dwelling type ratio, Flat or Apartment, Percentage of Dwellings',
  },
  dwellingOther: {
    name: 'Other',
    shortDescription: 'TODO',
    longDescription: 'TODO',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.MEDIAN,
    format: '0[.]0%',
    source: 'Dwelling type ratio, Other, Percentage of Dwellings',
  },
  jobsWalk: {
    name: 'Proportion of jobs accessible within 30 minutes by Walking',
    shortDescription: 'Proportion of jobs accessible within 30 minutes, Walking, % of all jobs in city',
    longDescription: 'TODO',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.MEDIAN,
    format: '0[.]0%',
    source: 'Proportion of jobs accessible within 30 minutes, Walking, % of all jobs in city',
  },
  jobsBike: {
    name: 'Jobs accessible within 30 minutes by bike',
    shortDescription: 'Jobs accessible within 30 minutes by bike',
    longDescription: 'TODO',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.MEDIAN,
    format: '0[.]0%',
    source: 'Proportion of jobs accessible within 30 minutes, By Bicycle, % of all jobs in city',
  },
  jobsCar: {
    name: 'Proportion of jobs accessible within 30 minutes by Car',
    shortDescription: 'Proportion of jobs accessible within 30 minutes, By Car, % of all jobs in city',
    longDescription: 'TODO',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.MEDIAN,
    format: '0[.]0%',
    source: 'Proportion of jobs accessible within 30 minutes, By Car, % of all jobs in city',
  },
  activeJourneys: {
    name: 'Active transport use',
    shortDescription: 'Active transport use, % of journeys to work',
    longDescription: 'TODO',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.MEDIAN,
    format: '0[.]0%',
    source: 'Active transport use, % of journeys to work',
  },
  publicJourneys: {
    name: 'Public transport use',
    shortDescription: 'Public transport use, % of journeys to work',
    longDescription: 'TODO',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.MEDIAN,
    format: '0[.]0%',
    source: 'Public transport use, % of journeys to work',
  },
};
