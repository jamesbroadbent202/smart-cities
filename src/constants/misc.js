export const DATA_URL = '/data/smartCitiesData.json';

export const NO_CATEGORY = 'overview';
export const NO_CITY = 'all-cities';

export const CARD_SIZES = {
  SMALL: 'small',
  LARGE: 'large',
};

export const INDICATOR_CARD_TYPES = {
  CATEGORY: 'CATEGORY',
  INDICATOR: 'INDICATOR',
};

// note, the values are lowercase because they're used in CSS class names
export const COLOR_NAMES = {
  CONTEXT: 'context',
  JOBS: 'jobs',
  HOUSING: 'housing',
  INFRASTRUCTURE: 'infrastructure',
  LIVEABILITY: 'liveability',
  INNOVATION: 'innovation',
  PLANNING: 'planning',
};

export const AGGREGATION_METHODS = {
  MEDIAN: 'MEDIAN',
  SUM: 'SUM',
};

export const DATA_TYPES = {
  NUMBER: 'NUMBER',
  TEXT: 'TEXT',
};

export const LINKS = {
  ORG: 'https://dta.gov.au/',
  CONTACT_US: 'mailto:performance-dashboard@digital.gov.au',
};

export const STRINGS = {
  SUB_CATS_INCLUDE: 'Sub categories include:',
};
