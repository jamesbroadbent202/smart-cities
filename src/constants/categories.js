import { COLOR_NAMES } from './misc';

export default [
  {
    id: 'jobs',
    colorName: COLOR_NAMES.JOBS,
    name: 'Jobs & Skills',
    description: 'Jobs and Skills encompasses all key elements of employment and training in our cities, including the performance of the employment market and the skill level of the workforce.',
    subCategories: [
      {
        name: 'Employment',
        tint: '040',
        shade: '600',
        indicatorIds: [
          'participationRate',
        ],
      },
      {
        name: 'Education',
        tint: '060',
        shade: '700',
        indicatorIds: [
          'educationalAttainmentYr12',
        ],
      },
    ],
    heroIndicatorId: 'population',
  },
  {
    id: 'housing',
    colorName: COLOR_NAMES.HOUSING,
    name: 'Housing',
    description: 'Housing encompasses three broad dimensions: the affordability of housing in our cities; the supply and diversity of new housing stock; and where housing is located, including how accessible it is to jobs and services.',
    subCategories: [
      {
        name: 'Buying houses',
        tint: '040',
        shade: '600',
        indicatorIds: [
          'jobAccessibility',
        ],
      },
      {
        name: 'Living in houses',
        tint: '060',
        shade: '700',
        indicatorIds: [
          'jobAccessibility',
        ],
      },
    ],
    heroIndicatorId: 'populationGrowth',
  },
  {
    id: 'infrastructure',
    colorName: COLOR_NAMES.INFRASTRUCTURE,
    name: 'Infrastructure & Other Things',
    navName: 'Infrastructure',
    description: 'Infrastructure and Investment encompasses all key dimensions of the cityâ€™s investment environment, with a particular focus on the quality, efficiency and effectiveness of infrastructure.',
    subCategories: [
      {
        name: 'Getting to work',
        tint: '040',
        shade: '600',
        indicatorIds: [
          'jobAccessibility',
        ],
      },
      {
        name: 'Getting around',
        tint: '060',
        shade: '700',
        indicatorIds: [
          'activeJourneys',
          'publicJourneys',
        ],
      },
      {
        name: 'Infrastructure',
        tint: '080',
        shade: '800',
        indicatorIds: [],
      },
    ],
    heroIndicatorId: 'population',
  },
  {
    id: 'liveability',
    colorName: COLOR_NAMES.LIVEABILITY,
    name: 'Liveability',
    description: 'Liveability and Sustainability encompasses three broad dimensions: the health and wellbeing of residents; the attractiveness and amenity of the city; and the state of the environment and the local response to climate change.',
    subCategories: [
      {
        name: 'Environment',
        tint: '040',
        shade: '600',
        indicatorIds: [],
      },
      {
        name: 'Public safety',
        tint: '060',
        shade: '700',
        indicatorIds: [],
      },
      {
        name: 'Life and death',
        tint: '080',
        shade: '800',
        indicatorIds: [],
      },
      {
        name: 'Socioeconomic disadvantage',
        tint: '100',
        shade: '900',
        indicatorIds: [],
      },
    ],
    heroIndicatorId: 'dispersion',
  },
  {
    id: 'innovation',
    colorName: COLOR_NAMES.INNOVATION,
    name: 'Innovation',
    description: 'Innovation and Digital Opportunities encompasses three broad dimensions: city productivity; innovation and entrepreneurship; and access to public data',
    subCategories: [
      {
        name: 'Data',
        tint: '040',
        shade: '600',
        indicatorIds: [],
      },
      {
        name: 'Working in digital',
        tint: '060',
        shade: '700',
        indicatorIds: [],
      },
    ],
    heroIndicatorId: 'population',
  },
  {
    id: 'planning',
    colorName: COLOR_NAMES.PLANNING,
    name: 'Planning',
    description: 'Governance, Planning and Regulation encompasses land use planning in cities and its administration, as well as how effectively local governance and regulation support economic, social and environmental outcomes.',
    subCategories: [
      {
        name: 'Development',
        tint: '040',
        shade: '600',
        indicatorIds: [],
      },
      {
        name: 'Population',
        tint: '060',
        shade: '700',
        indicatorIds: [
          'jobAccessibility',
        ],
      },
    ],
    heroIndicatorId: 'population',
  },
];
