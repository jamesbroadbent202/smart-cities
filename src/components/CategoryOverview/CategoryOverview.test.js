import React from 'react';
import { shallow } from 'enzyme';
import cloneDeep from 'lodash/cloneDeep';
import CategoryOverview from './CategoryOverview';
import { CATEGORY_IDS } from '../../constants';

jest.mock('../Icon/Icon', () => 'Icon');
jest.mock('../Card/IndicatorCard/IndicatorCard', () => 'IndicatorCard');
jest.mock('../Pill/Pill', () => 'Pill');

const defaultProps = {
  category: {
    id: 'jobs',
    name: 'Jobs and Skills',
    shortDescription: 'The short description',
    colorName: 'jobs',
    overviewIndicatorIds: [
      'population'
    ],
    iconId: 'infrastructureGettingToWork',
    subCategories: [
      { name: 'Employment' },
      { name: 'Education' },
    ],
  },
  city: {
    id: 'perth',
    name: 'Perth',
    indices: {
      population: 77,
    }
  },
};

it('should match Snapshot', () => {
  const component = shallow(
    <CategoryOverview {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});

it('should render the sub categories', () => {
  const component = shallow(
    <CategoryOverview
      {...defaultProps}
    />
  );

  const subCategoryLinks = component.find('.subCategoryLink');

  expect(subCategoryLinks.at(0).text()).toBe('Employment');
  expect(subCategoryLinks.at(1).text()).toBe('Education');
});

describe('when rendering the "context" category', () => {
  const propsForContextCategory = cloneDeep(defaultProps);
  propsForContextCategory.category.id = CATEGORY_IDS.CONTEXT;
  propsForContextCategory.category.colorName = CATEGORY_IDS.CONTEXT;

  const component = shallow(
    <CategoryOverview
      {...propsForContextCategory}
    />
  );

  it('should apply a modifier class', () => {
    expect(component.find('.categoryWrapper').hasClass('category__context')).toBe(true);
  });

  it('should not render a category icon', () => {
    expect(component.find('.categoryIcon').length).toBe(0);
  });

  it('should render a solid arrow icon in the category link', () => {
    expect(component.find('.categoryLinkPill Icon').prop('icon')).toBe('rightArrowInCircle');
  });

  it('should render the appropriate background color', () => {
    expect(component.find('.categoryWrapper').prop('style').background).toBe('WHITE');
  });

  it('should render the appropriate category link style', () => {
    expect(component.find('.categoryLinkPill').prop('style')).toEqual({
      background: 'WHITE',
      border: '2px solid CONTEXT_500',
      color: 'GREY_700',
    });
  });
});

describe('when rendering a category other than "context"', () => {
  const component = shallow(
    <CategoryOverview
      {...defaultProps}
    />
  );

  it('should not apply a modifier class', () => {
    expect(component.find('.categoryWrapper').hasClass('category__context')).toBe(false);
  });

  it('should render a category icon', () => {
    expect(component.find('.categoryIcon').length).toBe(1);
    expect(component.find('.categoryIcon Icon').prop('icon')).toBe('infrastructureGettingToWork');
  });

  it('should render an inverted arrow icon in the category link', () => {
    expect(component.find('.categoryLinkPill Icon').prop('icon')).toBe('rightArrowInCircleInverted');
  });

  it('should render the appropriate background color', () => {
    expect(component.find('.categoryWrapper').prop('style').background).toBe('JOBS_040');
  });

  it('should render the appropriate category link style', () => {
    expect(component.find('.categoryLinkPill').prop('style')).toEqual({
      background: 'JOBS_500',
      border: '2px solid JOBS_500',
      color: 'WHITE',
    });
  });
});

describe('when an array of cities is passed in', () => {
  const propsWithCities = cloneDeep(defaultProps);
  delete propsWithCities.city;
  propsWithCities.cities = [
    {
      id: 'perth',
      name: 'Perth',
      indices: {
        population: 77,
      }
    },
    {
      id: 'sydney',
      name: 'Sydney',
      indices: {
        population: 23,
      }
    },
  ];

  const component = shallow(
    <CategoryOverview {...propsWithCities} />
  );

  it('should aggregate data for all cities', () => {
    expect(component.find('IndicatorCard').prop('value')).toEqual([23, 77]);
  });

  it('should create a link to the all cities category page', () => {
    expect(component.find('.categoryLinkWrapper').prop('to')).toBe('/all-cities/jobs');
  });

});

describe('when a single city is passed in', () => {
  const component = shallow(
    <CategoryOverview {...defaultProps} />
  );

  it('should aggregate data for all cities', () => {
    expect(component.find('IndicatorCard').prop('value')).toBe(77);
  });

  it('should create a link to the all cities category page', () => {
    expect(component.find('.categoryLinkWrapper').prop('to')).toBe('/perth/jobs');
  });
});
