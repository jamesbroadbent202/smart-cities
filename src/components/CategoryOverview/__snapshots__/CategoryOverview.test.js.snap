// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<div className=\\"categoryWrapper\\" style={{...}}>
  <div className=\\"category\\">
    <div className=\\"categoryIcon\\">
      <Icon icon=\\"infrastructureGettingToWork\\" color=\\"JOBS_500\\" />
    </div>
    <div className=\\"textAndIndicatorWrapper\\">
      <div className=\\"categoryTextWrapper\\">
        <h2 className=\\"categoryTitle\\">
          Jobs and Skills
        </h2>
        <p className=\\"categoryDescription\\">
          The short description
        </p>
        <p className=\\"subCategoryWrapperText\\">
          Sub categories include:
        </p>
        <div>
          <NavLink to=\\"/perth/jobs#sc-employment\\" activeClassName=\\"active\\" ariaCurrent=\\"true\\">
            <Pill className=\\"subCategoryLink\\" shadow={true}>
              Employment
            </Pill>
          </NavLink>
          <NavLink to=\\"/perth/jobs#sc-education\\" activeClassName=\\"active\\" ariaCurrent=\\"true\\">
            <Pill className=\\"subCategoryLink\\" shadow={true}>
              Education
            </Pill>
          </NavLink>
        </div>
      </div>
      <div className=\\"indicatorCardAndLink\\">
        <IndicatorCard className=\\"indicatorCard\\" color=\\"JOBS_500\\" colorName=\\"jobs\\" indicator=\\"population\\" value={77} showPerformanceIndicator={false} />
        <NavLink to=\\"/perth/jobs\\" className=\\"categoryLinkWrapper\\" activeClassName=\\"active\\" ariaCurrent=\\"true\\">
          <Pill className=\\"categoryLinkPill\\" style={{...}} height={30}>
            <span className=\\"categoryLinkText\\">
              Jobs and Skills
               section
            </span>
            <Icon icon=\\"rightArrowInCircleInverted\\" size={20} color=\\"JOBS_500\\" />
          </Pill>
        </NavLink>
      </div>
    </div>
  </div>
</div>"
`;
